src/Data/Key.hs

```diff
@@ -1114,16 +1119,31 @@ instance ZipWithKey Seq where
   zipWithKey f a b = Seq.zipWith id (Seq.mapWithIndex f a) b
 
 instance Adjustable Seq where
-  adjust = Seq.adjust
 
+  adjust = 
+#if MIN_VERSION_containers(0,5,8)
+           Seq.adjust' -- Use the prefered strict version when available
```

Comment:
Are there semantic differences between `adjust'` and `adjust`, or is one simply more efficient space-wise than the other? If it's the former, then we're putting a conditional API into place, which scares me a little bit...

Reply:
The same thought crossed my mind. If you look at the documentation for [`adjust`](http://hackage.haskell.org/package/containers/docs/Data-Sequence.html#v:adjust), it says:
>  adjust can lead to poor performance and even memory leaks, because it does not force the new value before installing it in the sequence. adjust' should usually be preferred.

If you look at the source of `adjust` and `adjust'`, the latter uses a `ForceBox` to force the spine of the `Sequence`.

I imagine the creates a semantic difference in some situations for the adjusted `Sequence` that is returned from the `adjust` call, but there is no difference for the polymorphic `a` value. However, I think that users would like this semantic difference if it means avoiding a space leak, which is why I put the conditional definition in the pull request. It seemed preferable to dropping support for GHC 7.10 & GHC 8.0, and unconditionally calling `adjust'`.

Reply:
Why not just implement `adjust'` like so on older versions of GHC?

```haskell
adjust' f i xs =
 case xs !? i of
   Nothing -> xs
   Just x -> let !x' = f x
             in update i x' xs
```

(This was taken from the [Haddocks for `adjust'`](https://hackage.haskell.org/package/containers-0.6.0.1/docs/Data-Sequence.html#v:adjust-39-).)

It's not as efficient as it could be, of course, but I'm not going to lose sleep over the code paths for old GHCs being slow.

Reply:
This is a much better idea, creating consistent API semantics across GHC versions. Great idea.

I have updated the pull request to incorporate the code suggestion above!